src/routes/attendanceSheets.js

```diff
@@ -0,0 +1,41 @@
+'use-strict';
+
+const Joi = require('joi');
+Joi.objectId = require('joi-objectid')(Joi);
+const { list, create } = require('../controllers/attendanceSheetController');
+const { formDataPayload } = require('./validations/utils');
+
+exports.plugin = {
+  name: 'routes-attendanceSheets',
+  register: async (server) => {
+    server.route({
+      method: 'GET',
+      path: '/',
+      options: {
+        auth: { scope: ['courses:read'] },
+        validate: {
+          query: Joi.object({ course: Joi.objectId() }),
+        },
+      },
+      handler: list,
+    });
+
+    server.route({
+      method: 'POST',
+      path: '/',
+      options: {
+        payload: formDataPayload(),
+        validate: {
+          payload: Joi.object({
+            course: Joi.objectId().required(),
+            file: Joi.any().required(),
+            trainee: Joi.objectId(),
+            date: Joi.date(),
+          }),
```

Comment:
tu peux preciser qu'il faut a trainee ou date obligatoire

Reply:
je ne trouve pas comment vérifier le type de la formation, ni ici pour le payload ni dans le model, j'aimerais écrire quelquechose comme
`.when('course.type', { is: INTRA, then: Joi.required() })`
mais course n'est qu'un id...
et dans le model ce serait 
`  date: { type: Date,  required() { return this.course.type === INTRA; }, },`

Reply:
tu peux le faire dans un preHandler, non ?